skillmap/src/components/HeaderBar.tsx

```diff
@@ -60,38 +83,55 @@ export class HeaderBarImpl extends React.Component<HeaderBarProps> {
         return items;
     }
 
+    protected getUserMenu() {
+        const { signedIn } = this.props;
+        const items = [];
+        const user = pxt.auth.client()?.getState().profile;
+
+        if (signedIn) {
+            items.push({
+                id: "signout",
+                label: lf("Sign Out"),
+                onClick: this.onLogoutClicked
+            })
+        }
+
+        const avatarElem = user?.idp?.picture?.dataUrl
+            ? <div className="avatar"><img src={user?.idp?.picture?.dataUrl} alt={lf("User Menu")}/></div>
+            : undefined;
+
+        const initialsElem = user?.idp?.displayName
+            ? <span className="circle">{pxt.auth.userInitials(user?.idp?.displayName)}</span>
+            : undefined;
+
+        return <div className="user-menu">
+            {signedIn
+             ? <Dropdown icon="star" items={items} picture={avatarElem || initialsElem} className="header-dropdown"/>
+             : <HeaderBarButton className="sign-in" icon="xicon icon cloud-user" title={lf("Sign In")} label={lf("Sign In")} onClick={this.props.dispatchShowLoginModal}/>}
+        </div>;
+    }
+
     render() {
-        const { activityOpen, currentActivityDisplayName } = this.props;
-        const logoAlt = "MakeCode Logo";
-        const organizationLogoAlt = "Microsoft Logo";
-        const logoSrc = (isLocal() || !pxt.appTarget?.appTheme?.logoUrl ) ? resolvePath("assets/logo.svg") : pxt.appTarget?.appTheme?.logo;
+        const { activityOpen } = this.props;
+        const hasIdentity = pxt.auth.hasIdentity();
 
+        const appTheme = pxt.appTarget?.appTheme;
         const settingItems = this.getSettingItems();
         const helpItems = this.getHelpItems();
 
         return <div className="header">
             <div className="header-left">
-                <div className="header-logo">
-                    <img src={logoSrc} alt={logoAlt} />
-                </div>
-                { activityOpen ?
-                    <HeaderBarButton icon="icon arrow left" label={lf("Back")} title={lf("Return to activity selection")} onClick={this.onBackClicked}/> :
-                    <HeaderBarButton icon="icon home" label={lf("Home")} title={lf("Return to the editor homepage")} onClick={this.onHomeClicked}/>
-                }
+                {this.getOrganizationLogo(appTheme)}
+                {this.getTargetLogo(appTheme)}
             </div>
-            { currentActivityDisplayName &&
-                <div className="header-activity-display-name" title={currentActivityDisplayName}>
-                    {currentActivityDisplayName}
-                </div>
-            }
+
             <div className="spacer" />
             <div className="header-right">
+                {activityOpen && <HeaderBarButton icon="icon arrow left" title={lf("Return to activity selection")} onClick={this.onBackClicked}/>}
```

Comment:
Might want to get @Jaqster or @kiki-lee to review this text to make sure we're consistent on what the activity selection screen is called

Reply:
Text is unchanged from previous version, just moved downward. @livcheerful was this an intentional change?

Reply:
Intentional how? Moving downwards is part of refreshing skillmap nav bars to align with the rest of the aoo

Reply:
Oh got it! I was wondering about the removal of currentActivityDisplayName (to @shakao's comment).